### ISSUE TITLE
deps.sh fails when running in a path with spaces

### ISSUE THREAD

**Issue by jmpfar** [libjxl/libjxl#3684]
**Describe the bug**
When running `deps.sh` in a path with spaces, the dependencies are downloaded to the wrong path: e.g.

```bash
❯ pwd
/Users/hello/code/test/te st/blah/libjxl-0.10.3/build

❯ ./deps.sh
fatal: not a git repository (or any of the parent directories): .git
Downloading testdata version 873045a9c42ed60721756e26e2a6b32e17415205...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
[...]

❯ ls third_party/skcms

❯ ls third_party/highway

❯ ls ../../../../test$'\n'st/blah/libjxl-0.10.3/
downloads   testdata    third_party
```

The files are instead in a directory that contains a `\n` character

**To Reproduce**
Create a path containing a space, extract libjxl source code in a directory under it and runs `./deps.sh`

**Expected behavior**
Dependencies will be downloaded under `third_party`

**Screenshots**

**Environment**
 - OS: MacOS 14.5
 - Compiler version: Apple clang version 15.0.0
 - CPU type: Apple Sillicon M2
 - cjxl/djxl version string: 0.10.3

**Additional context**

**Comment by jmpfar** [comment]
At least for me, what fixed the issue was changing:
https://github.com/libjxl/libjxl/blob/36e05143fb13a15dc5cf73c8b8861a7e20859975/deps.sh#L11-L13

to
```bash
MYDIR=$(dirname "$(realpath "$0")")
```

**Comment by mo271** [comment]
Thanks, should be fixed by #3698
